data_ingestion/main.py

```diff
@@ -56,17 +52,40 @@ def ingest_data(event, context):
   notify_data_ingested_topic = os.environ['NOTIFY_DATA_INGESTED_TOPIC']
 
   if id == _HOUSEHOLD_INCOME:
-    census.upload_household_income(url, gcs_bucket, filename)
-  elif id == _STATE_NAMES:
-    census.upload_state_names(url, gcs_bucket, filename)
+    if 'url' not in event_dict or 'filename' not in event_dict:
```

Comment:
Feels like there should be a better way to do this section to not duplicate all this boilerplate. I think separating out the parsing of data source metadata from handling the logging would help, but probably not worth bothering to do that for this PR, so no need to change anything

Reply:
Yeah definitely a broader conversation is needed and may depend on on the actual data sources.

I do think some of this can be abstracted in a way that makes it more flexible to changing the pubsub setup later (eg by parsing the pubsub into a SourceMetadata class and then passing that through to helper functions so you don't need to branch on filename vs file prefix or other attribute differences). But that's more of a "nice-to-have", not something we really need :P